src/DependencyInjection/CsaGuzzleExtension.php

```diff
@@ -132,4 +133,20 @@ function ($tags) { return $tags[0]['alias']; },
             return !isset($explicitlyConfiguredIds[$name]) || $explicitlyConfiguredIds[$name];
         });
     }
+
+    private function buildGuzzleConfig($options)
+    {
+        if (!isset($options['config'])) {
+            return null;
```

Comment:
Also, I would actually move this part up in the previous method (`processClientsConfiguration()`), in order to have less code, and directly inject the `$config` in the `buildGuzzleConfig()` method.

i.e.

```php
if (isset($options['config'])) {
    $client->replaceArgument(0, $this->buildGuzzleConfig($options['config']));
}

// ...

public function buildGuzzleConfig(array $config)
{
    // ...
}
```

This way, you can also ensure that `$config` key is an array.